RELEASING.md

```diff
@@ -1,3 +1,69 @@
+
+### Release Phase 1 - Preparation
+
+To track the major / minor / patch release, a github issue will be created. The issue will be created on the OpenSearch Dashboards repository for major / minor releases, with the overall issue tracked on the OpenSearch build repository. Only an overall issue will be created on the OpenSearch build repository for patch release.
+
+(Example github issues of [major](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1548) / [minor](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2230) / [patch](https://github.com/opensearch-project/opensearch-build/issues/2650) release)
+
+The release manager (RM) will be assigned to the OpenSearch Dashboards release issue and will be responsible for reviewing all tasks in this issue for preparation. It's also a good idea to compare the current release issue to the issue of previous release version at this point to ensure that all new processes have been captured.
+
+RM will review the [public roadmap](https://github.com/orgs/opensearch-project/projects/1) and confirm the release scope with other OpenSearch Dashboards [maintainers](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/MAINTAINERS.md) as well as feature owners. All features not ready for current release will be relabeled. For example, if current release version is v2.3.0, all features not ready will be labeled as v2.4.0. (or even later by discussing with the feature owner). After then, RM will need to check all PRs for the current release version are merged into main branch and backported PRs are merged with all CI passed.
+
+#### Prepare BWC data and update BWC versions
+
+Backwards Compatibility Tests (BWC) are cypress tests that verify any changes are backwards compatible with previous versions. RM will generate test data and test locally following instructions [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/TESTING.md#backwards-compatibility-tests) and cut PR to include both generated data and version upgrade for automated build. (See example [PR](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2393/files))
+
+#### Cut release branch for major / minor release
+
+For major / minor release, RM will need to cut the release branch from the parent branch, [following OpenSearch project branching](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#opensearch-branching)
+
+### Release Phase 2 - Pre-Release
+
+The release process for OpenSearch is centralized. Jenkins workflows are in place to run daily snapshot builds for OpenSearch and OpenSearch Dashboards. RM will update the release branch version in the distribution manifest (see example [PR](https://github.com/opensearch-project/opensearch-build/pull/2586/files)) and increment the parent branch version (see example [PR](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2295/files)).
```

Comment:
NIT: `daily` snapshots aren't build. As far as I am aware, and I could be wrong but the current workflow actually is a CRON job checks if there is a diff in the components. Depending on how close the release is for the version the CRON job is more frequent going from like 6 hours to like 2 hours or something. And if there aren't changes then there is no build.